Show logs when no step name is provided

ncskier opened the pull request:
# Changes
Fixes #709
A TaskRun has all of the information about the steps in its status
(including the default step names when no name is provided). So, I
changed the TaskRun container and PipelineRun component to only use the
TaskRun's status for retreiving information about the TaskRun
(previously they were using the Task definition as well).

Here's an example of the default step names when no name is provided:
<img width="862" alt="Screen Shot 2020-01-08 at 2 18 45 PM" src="https://user-images.githubusercontent.com/16961496/72008454-dcec3a80-3221-11ea-9a1d-34ab4a89b1bd.png">


I also updated the TaskTree's Storybook entry to be more descriptive.

/cc @AlanGreene 
/cc @a-roberts 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

AlanGreene commented on packages/components/src/components/PipelineRun/PipelineRun.js:
```diff
@@ -118,7 +118,7 @@ export /* istanbul ignore next */ class PipelineRunContainer extends Component {
           taskRun = theRun; // eslint-disable-line no-param-reassign
         } else {
-          steps = stepsStatus(task.spec.steps, taskRun.status.steps);
+          steps = stepsStatus(taskRun.status.steps, taskRun.status.steps);
```
I remember now why we used the task spec

we may not have status for all steps, so would end up leaving some steps out of the list until they begin executing
using the task spec we get the full list of steps even if we don't have status for them yet

ncskier replied:
Hm, in what case were you seeing that happen?

When I run a Pipeline, all of the steps seem to appear even before they begin executing:
<img width="432" alt="Screen Shot 2020-01-08 at 4 05 22 PM" src="https://user-images.githubusercontent.com/16961496/72015949-c39eba80-3230-11ea-851d-ee7c7d874301.png">

AlanGreene replied:
yeah I'm not seeing it anymore either... ok let's go for it 👍

ncskier replied:
Ok, thanks @AlanGreene! The PR's ready for review now 🙂

AlanGreene replied:
@ncskier looks like there was another reason for `task.spec.steps` after all, it was used to populated the step definition data displayed on the details tab

with this change we're now showing essentially the same status information (only difference being the step id on the details tab) on both the status and details tabs

I think reverting this line here and in TaskRun should resolve this issue without affecting the display of logs when no step name is given. Testing it out now... 🤞

ncskier commented:
/retest

ncskier commented:
When working on this issue, I wanted to clean up some of the PipelineRun component and TaskRun container code; however, it is not within the scope of this issue. So, I've created issue https://github.com/tektoncd/dashboard/issues/885 to track the code clean up piece of work.

AlanGreene commented on packages/utils/src/utils/index.js:
```diff
@@ -112,6 +112,32 @@ export function stepsStatus(taskSteps, taskRunStepsStatus = []) {
+*/
+export function reorderSteps(unorderedSteps, orderedSteps) {
+  if (
+    typeof unorderedSteps === 'undefined' ||
```
is `!unorderedSteps || !orderedSteps` sufficient here?

ncskier replied:
Thanks Alan, I always forget that I can use that syntax with JavaScript 👍

AlanGreene commented on packages/utils/src/utils/index.js:
```diff
@@ -112,6 +112,32 @@ export function stepsStatus(taskSteps, taskRunStepsStatus = []) {
+    return [];
+  }
+  const steps = [];
+  orderedSteps.forEach(({ name }, idx) => {
```
can use `map` instead of `forEach` + `push`

ncskier replied:
Ooo, good point, nice!

AlanGreene commented on src/containers/TaskRun/TaskRun.js:
```diff
@@ -101,14 +106,17 @@ export /* istanbul ignore next */ class TaskRunContainer extends Component {
+
+    const { steps } = taskRun.status;
+    const stepDefinitions =
+      typeof taskRun.spec.taskSpec !== 'undefined'
```
can this be just `taskRun.spec.taskSpec`?

ncskier commented:
/test tekton-dashboard-unit-tests

ncskier commented:
/retest

ncskier commented:
It says the unit tests have been running for an hour 😓 
/test tekton-dashboard-unit-tests

ncskier commented:
Hmmm, looks like the tests worked now?

@AlanGreene thanks for the review! I've addressed your suggestions, so please take another look when you get the chance 🙂
